docs/docs/api-ref.md

```diff
@@ -5,257 +5,3010 @@ layout: docs
+
+**Example**
 
 ```py
-Sites.get()
+
+server.auth.sign_out()
+
+
 ```
 
-### Update Site
 
-Modifies a site. The site item object must include the site ID and overrides all other settings.
+
+
+**See Also**  
+[Sign in and Out](sign-in-out)  
+[Server](#server)
+
+<br>
+<br>
+
+
+
+
+## Connections
+
+The connections for Tableau Server data sources and workbooks are represented by a `ConnectionItem` class.  You can call data source and view methods to query or update the connection information.  The `ConnectionCredentials` class represents the connection information you can update. 
+
+### ConnectionItem class
 
 ```py
-Sites.update(site_item_object)
+ConnectionItem()
 ```
 
-### Delete Site
+**Attributes**  
+
+Name   |  Description     
+ :--- | : ---    
+`datasource_id` |  The identifier of the data source. 
+`datasource_name` |  The name of the data source.
+`id`  |  The identifer of the connection.
+`connection_type`  |  The type of connection. 
+
+
+Source file: models/connection_item.py  
+
+<br>
+<br>
+
 
-Deletes the site with the given ID.
+
+### ConnectionCredentials class
 
 ```py
-Sites.delete(id)
+ConnectionCredentials(name, password, embed=True, oauth=False)
 ```
 
-## Projects
 
-Source files: server/endpoint/projects_endpoint.py
+The `ConnectionCredentials` class corresponds to workbook and data source connections. 
+
+In the Tableau Server REST API, there are separate endopoints to query and update workbook and data source connections. 
+
+**Attributes**
+
+Attribute | Description
+:--- | :---
+`name`     | The username for the connection.
+`embed_password`  |  (Boolean) Determines whether to embed the passowrd (`True`) for the workbook or data source connection or not (`False`). 
+`password`  |  The password used for the connection.   
+`server_address`   |  The server address for the connection.   
+`server_port`   |  The port used by the server.  
+`ouath`  |  (Boolean) Specifies whether OAuth is used for the data source of workbook connection. For more information, see [OAuth Connections](https://onlinehelp.tableau.com/current/server/en-us/protected_auth.htm?Highlight=oauth%20connections){:target="_blank"}.  
+
+
+Source file: models/connection_credentials.py
 
-### Create Project
+<br>
+<br>
 
-Creates a project for the given project item object.
+## Data sources
+
+Using the TSC library, you can get all the data sources on a site, or get the data sources for a specific project. 
+The data source resources for Tableau Server are defined in the `DatasourceItem` class. The class corresponds to the data source resources you can access using the Tableau Server REST API. For example, you can gather information about the name of the data source, its type, and connections, and the project it is associated with. The data source methods are based upon the endpoints for data sources in the REST API and operate on the `DatasourceItem` class.  
```

Comment:
Um.. just two. But I changed it up. Just one now, but the sentence ended up with a bunch of "its"  :)